x-pack/plugins/ml/public/application/jobs/jobs_list/components/utils.js

```diff
@@ -180,44 +179,53 @@ function showResults(resp, action) {
 
 export async function cloneJob(jobId) {
   try {
-    const job = await loadFullJob(jobId);
-    if (job.custom_settings && job.custom_settings.created_by) {
+    const [cloneableJob, originalJob] = await Promise.all([
+      loadFullJob(jobId, true),
+      loadFullJob(jobId, false),
+    ]);
+    if (cloneableJob.custom_settings && originalJob?.custom_settings?.created_by) {
```

Comment:
if `custom_settings` does not exist in `cloneableJob` this entire block will be skipped.
To be safe we should only check for `originalJob?.custom_settings?.created_by` and then create `cloneableJob.custom_settings` if it doesn't exist.